udata_front/theme/gouvfr/datagouv-components/src/composables/organizations/useOrganizationType.ts

```diff
@@ -0,0 +1,50 @@
+import { computed, toValue, type MaybeRefOrGetter } from "vue";
+import type { Organization } from "../../types/organizations";
+
+export const PUBLIC_SERVICE = "public-service";
+export const ASSOCIATION = "Association";
+export const COMPANY = "Company";
+export const LOCAL_AUTHORITY = "Local authority";
+export const USER = "user";
+
+export type OrganizationTypes = typeof PUBLIC_SERVICE | typeof ASSOCIATION | typeof COMPANY | typeof LOCAL_AUTHORITY;
+
+export type UserType = typeof USER;
+
+/**
+ * Checks if the organization has the provided type
+ * @param organizationRef the organization
+ * @param type the organization type
+ */
+export function isType(organizationRef: MaybeRefOrGetter<Organization>, type: OrganizationTypes) {
+  return hasBadge(organizationRef, type);
+}
+
+/**
+ * Checks if the organization has the provided badge
+ * @param organizationRef the organization
+ * @param kind the badge
+ */
+export function hasBadge(organizationRef: MaybeRefOrGetter<Organization>, kind: string) {
+  return toValue(organizationRef).badges.some(badge => badge.kind === kind);
+}
+
+export default function useOrganizationType(organizationRef: MaybeRefOrGetter<Organization>) {
+  const isPublicService = computed(() => isType(organizationRef, PUBLIC_SERVICE));
+  const isAssociation = computed(() => isType(organizationRef, ASSOCIATION));
+  const isLocalAuthority = computed(() => isType(organizationRef, LOCAL_AUTHORITY));
+  const type = computed(() => {
+    if(isPublicService.value) {
+      return PUBLIC_SERVICE;
+    } else if(isAssociation.value) {
+      return ASSOCIATION;
+    } else if (isLocalAuthority.value) {
+      return LOCAL_AUTHORITY;
+    } else {
+      return COMPANY;
+    }
+  });
+  return {
+    type,
+  };
+}
```

Comment:
Same, I think it should be an helper `getOrganizationType(org: Organization)`  and the computed moved to the component to avoid leaking Vue concepts:
```
const organizationType = computed(() => getOrganizationType(org.value))
```